verification/verify.go

```diff
@@ -160,6 +171,32 @@ func IsRetryable(err error) bool {
 	return retryable
 }
 
+func (sv *SegmentVerifier) sigVerification(params *Params) error {
+	if params.Orchestrator == nil || params.Orchestrator.TicketParams == nil {
+		return nil
+	}
+
+	if len(params.Results.Segments) != len(params.Renditions) {
+		return errPMCheckFailed
+	}
+
+	segHashes := make([][]byte, len(params.Renditions))
+	for i := range params.Renditions {
+		segHashes[i] = crypto.Keccak256(params.Renditions[i])
+	}
+
+	// Might not have seg hashes if results are directly uploaded to the broadcaster's OS
+	// TODO: Consider downloading the results to generate seg hashes if results are directly uploaded to the broadcaster's OS
+	if !sv.verifySig(
```

Comment:
Just noting that this check will always fail when B is using an external OS since `Renditions` would be populated with a bunch of nil slices in that case. We can tackle that problem as a part of #1382.

Reply:
ok